### PR TITLE
bulk install python dependencies

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

bulk install python dependencies

## Fixes

https://github.com/fetchai/agents-aea/issues/2092

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`scripts/generate_ipfs_hashes.py`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

File: tests/test_cli/test_install.py [hunk 1]
```diff
@@ -94,10 +96,6 @@ def setup_class(cls):
                     "version": "==0.1.0",
                     "index": "https://test.pypi.org/simple",
                 },
-                "this_is_a_test_dependency_on_git": {
```
Comment (DavidMinarsch) [inline tests/test_cli/test_install.py:97]: Why removed?

**Review by DavidMinarsch — APPROVED** [review]
LGTM, one comment

File: tests/test_cli/test_install.py (hunk 1 above)
Comment (solarw) [reply tests/test_cli/test_install.py:97]: it used for test that should fail on the first dependency, but when i pass all dependencies to pip, pip asks for credentials for git, cause it tries to get access to this repo

Comment (DavidMinarsch) [reply tests/test_cli/test_install.py:97]: ^ yes that's worth trying. Although do we always want it to be installed in the same environment? What about the multiprocess case?

Comment (solarw) [reply tests/test_cli/test_install.py:97]: will check pip direct usage.

about venv - shouldn't be a problem, we can call any code inside venv, code is ready.


about venv support in MAM, it's not enabled yest as i remember, think it's better to ensure MAM multiprocess is completely ready - integrate with GUI

**Comment by solarw** [comment]
@DavidMinarsch 
I updated install dependencies with direct call of python code for pip install.

one point: now i can not control pip install execution time.
but pip has own timeout for sockets and own retries mechanism

anyway commit can be reverted easily

**Comment by solarw** [comment]
there is a weird test crash. PR market as WIP  for an invetigation

**Comment by solarw** [comment]
this pr is ready,
moved experimetnal code to another branch: https://github.com/fetchai/agents-aea/pull/2619

**Review by marcofavorito — APPROVED** [review]

**Review by 5A11 — APPROVED** [review]
lgtm

**Pull request merged**